clients/drcachesim/tools/schedule_stats.cpp

```diff
@@ -330,7 +330,13 @@ schedule_stats_t::print_percentage(double numerator, double denominator,
 void
 schedule_stats_t::print_counters(const counters_t &counters)
 {
-    std::cerr << std::setw(12) << counters.threads.size() << " threads\n";
+    std::cerr << std::setw(12) << counters.threads.size() << " threads";
+    if (!counters.threads.empty()) {
+        std::cerr << ":";
+        for (auto thread : counters.threads)
+            std::cerr << " " << thread;
```

Comment:
Would comma-separated be nicer?

Reply:
Hmm, maybe.  I was dealing with thread lists from another source which were just space-separated.  Comma might be easier to auto-parse.  Commas when iterating a set do require a slightly awkward loop, checking end inside the loop.  I guess we can live with that.  OK, done.